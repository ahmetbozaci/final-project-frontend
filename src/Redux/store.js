import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './rootReducer';
import fetchDoctors from './doctor/doctorActions';

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk, logger)),
);
store.dispatch(fetchDoctors());

export default store;
