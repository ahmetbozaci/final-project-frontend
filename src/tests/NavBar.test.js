import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Navbar from '../components/NavBar';

it('Navbar renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Router><Navbar /></Router></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
