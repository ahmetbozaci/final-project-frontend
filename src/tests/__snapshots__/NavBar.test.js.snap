// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="navbar nav-mobile navbar-expand navbar-dark bg-primary fixed-left"
>
  <img
    alt="recovon logo"
    className="nav-logo mx-auto"
    src="recovon-logos-transparent.png"
  />
  <h2
    className="nav-mobile-logo"
  >
    Recovon
  </h2>
  <div
    className="collapse navbar-collapse nav-element-space"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="username p-lg-2 pb-lg-4"
      >
        Welcome
         
      </li>
      <li
        className=""
      >
        <a
          className="nav-link nav-element"
          href="/doctors"
          onClick={[Function]}
        >
          Doctors
        </a>
      </li>
      <li
        className=""
      >
        <a
          className="nav-link nav-element"
          href="/myappointments"
          onClick={[Function]}
        >
          My Appointments
        </a>
      </li>
      <li
        className="nav-logout-space"
      >
        <form
          onSubmit={[Function]}
        >
          <button
            className="btn btn-secondary"
            onClick={[Function]}
            type="submit"
          >
            Logout
          </button>
        </form>
      </li>
    </ul>
  </div>
</nav>
`;
